### ISSUE TITLE
VST3: cannot set parameters on Reaper/Linux

### ISSUE THREAD

**Issue by lucianoiam** [DISTRHO/DPF#343]
Since this plugin format needs some work on Mac and Windows I started by testing it on Linux which is probably the reference platform; VST3 Parameters example works great on Carla indeed.

On Reaper/Linux found the following (again in Parameters):

- UI works as expected
- Updating a parameter using Reaper generic UI is reflected next time UI is opened
- Updating a parameter from the custom plugin UI does not have any effect in the generic UI

In other words VST3 plugins cannot update parameters for the Reaper host. First spotted this in a plugin of my own.

I debugged up to `requestParameterValueChange()`  (DistrhoPluginVST3.cpp) and all seems good, `begin_edit` and `perform_edit` returning `V3_TRUE`.

Any hint on how to further debug this?

**Comment by lucianoiam** [comment]
That is a good idea, thanks. I need to read how VST3 works as well lol.

**Comment by falkTX** [comment]
This should be fixed now, confirmation would be most welcome

**Comment by lucianoiam** [comment]
Sadly it still seems broken, just tested c2f66ac on Reaper/Mac:

1. Open d_parameters.vst3 UI
2. Click top-left (make it orange)
3. Switch to Reaper generic UI
4. Top-left is zero

**Comment by MARABOTTO** [comment]
VST3 is also broken on Ableton (can see the plugin but can't load it). :(

**Comment by falkTX** [comment]
we need to check this again after latest vst3 fixes in dpf.

**Comment by falkTX** [comment]
and it is still broken.

it behaves properly in other hosts like Carla and FLStudio though.

**Comment by falkTX** [comment]
Should be fixed with latest code now, I verified with https://github.com/trummerschlunk/master_me that things generally seem to work there.

**Comment by lucianoiam** [comment]
Just repeated this on Reaper+Linux and it now works great; maybe it is time to close the ticket.

> 
>     1. Open d_parameters.vst3 UI
> 
>     2. Click top-left (make it orange)
> 
>     3. Switch to Reaper generic UI
> 
>     4. Top-left is zero

**Comment by falkTX** [comment]
I suspected as such, thanks for confirming!
